pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -96,63 +109,137 @@ func (e *DockerRegistryServiceController) Stop() {
 	}
 }
 
-func (e *DockerRegistryServiceController) getServiceLocation(service *api.Service) string {
+// waitForSync polls to see if all the controllers are synced.  If so, it then makes a call to
+// update the dockercfg secrets initially
+func (e *DockerRegistryServiceController) waitForSync(stopCh <-chan struct{}) {
+	for {
+		select {
+		case <-stopCh:
+			return
+		case <-time.Tick(100 * time.Millisecond):
```

Comment:
use `time.After`. `time.Tick` sets up a ticker that can never be stopped or garbage collected.